### DISCUSSION TITLE
希望可以独立调节app音量

### DISCUSSION THREAD

**Discussion by SolidRainHolder** [xfangfang/wiliwili#153] [Ideas]
希望可以独立调节app音量，现在只使用了系统音量，但是我有时候需要发声的app不止一个，不过我是mac电脑，目前主机上没这个需求，觉得电脑包还是挺需要的。

**Comment by xfangfang** [comment]
最新测试版完成了这个功能，可以去测试看看：https://nightly.link/xfangfang/wiliwili/workflows/build.yaml/dev

**Reply by SolidRainHolder** [reply]
感谢作者的工作，很高兴建议能被采纳；
暂时用win电脑小试了一下，**重启app时，上次音量调节之后位置，没有被记忆**，切换视频倒是没有这个问题。
回头我再细测下，之后有什么问题还给您反馈

**Reply by xfangfang** [reply]
@SolidRainHolder 不用客气，重启不记忆音量是因为没有做这个功能，我还没考虑好要不要做这个。

**Reply by SolidRainHolder** [reply]
整理了一下，关于wiliwili我的一些想法
### 1、重启记忆音量
我认为是有必要的，既然调音量了，就说明视频音量是不合适的，这该归类在用户设置里，只是调节的比较频繁罢了。以照我的使用场景，每次打开第一件事就需要先调一下音量，系统音量基本上不会去动的。考虑到作为掌机用户的使用习惯是直接按机器上的音量键，这其实是不矛盾的，本身也不会去调app的音量，而且，使用t组按键调节音量（第二条建议），比掌机音量按钮要来的方便得多，不管是switch还是steamdeck，况且switch主机模式的时候，更难调音量。我相信这是能改变用户习惯的功能
### 2、以xbox手柄为例，希望在视频播放界面的时候，rt和lt能够调节音量
现在视频播放界面的逻辑是
视频小窗播放时
选择到视频窗口时，rb，lb调节播放进度，不能跳转到评论区，rt，lt则在视频小窗和评论区模块进行跳转，但跳到评论区以后便无法跳回到视频小窗
视频全屏播放时
由于没有评论区了，rt，lt就没有作用了，rb，lb依然可以调节视频进度

建议可以改为以下逻辑
视频小窗播放时
十字键，选择到视频小窗时，b组键依然负责调节进度，t组负责调节音量，十字键用来跳转到评论区或者跳回来，这样逻辑就一致了，保证bt组永远不要跳出同层级标签

视频全屏播放时
b组按键功能不变，t组按键调节音量

策划一张嘴系列建议（您可以不用当真）
1、使用t组或者b组调节进度和音量时，没必要显示播放的ui，在后台调节就好（当然优化做法是可以加个会自动隐藏的小的ui反馈）。
在专注欣赏内容时，我不需要知道具体到哪儿了，可能只是当前内容快一点或者慢一点，音量同理，我不需要知道具体现在是多少音量，只是听着合适就行，而且跳播放ui出来有时候会出现遮挡字幕当问题，显示隐藏ui专注让y键去做即可
保留打开ui时用十字键选择音量再调节具体值的功能

2、希望评论区引用的b站视频链接可以选择跳转到该视频，但牵扯到非b站链接大概会比较麻烦？

内容没啥逻辑性想到哪儿写哪儿，希望wiliwili越来越好，看视频也能有游戏性哈哈

**Reply by xfangfang** [reply]
@SolidRainHolder 我把音量保存起来了；至于使用 lt/rt 来调节音量，我感觉不是很合适。

可能比较容易误触，尤其是 switch那个手柄，总是会不小心误触到，所以一直比较避免使用。

我认为比较理想的模式是只有 `方向键/确认/返回/和一个额外的功能键` 就像电视上一样，这样我感觉会比较方便新人使用，目前除了跳转进度和搜索没有办法做到，其他都可以做到，这两个我还没想好怎么好好调整一下。
